CycloneDX/Services/ProjectAssetsFileService.cs

```diff
@@ -95,51 +92,13 @@ public HashSet<NugetPackage> GetNugetPackages(string projectFilePath, string pro
 
             return packages;
         }
-        public bool SetIsDirectReference(string packageName, string targetRuntime, JsonElement jsonContent)
+        public bool SetIsDirectReference(LibraryDependency dependency)
         {
-            string framework = TargetFrameworkToAlias(targetRuntime);
-            JsonElement packageProperties;
-            if (jsonContent.GetProperty(framework).GetProperty("dependencies").TryGetProperty(packageName, out packageProperties))
-            {
-                // every direct reference has target property
-                return packageProperties.TryGetProperty("target", out _);
-            }
-            return false;
+            return dependency?.ReferenceType == LibraryDependencyReferenceType.Direct;
         }
-        public bool SetIsDevDependency(string packageName, string targetRuntime, JsonElement jsonContent)
+        public bool SetIsDevDependency(LibraryDependency dependency)
         {
-            string framework = TargetFrameworkToAlias(targetRuntime);
-            JsonElement packageProperties;
-            if (jsonContent.GetProperty(framework).GetProperty("dependencies").TryGetProperty(packageName, out packageProperties))
-            {
-                // suppressParent: exists only for development dependencies
-                return packageProperties.TryGetProperty("suppressParent", out _);
-            }
-            return false;
-        }
-
-        /// <summary>
-        /// Converts an asset file's target framework value into a csproj target framework value.
-        ///
-        /// Examples:
-        ///     .NetStandard,Version=V3.1 => netstandard3.1
-        ///     .NetCoreApp,Version=V3.1  => netcoreapp3.1
-        ///     netcoreapp3.1  => netcoreapp3.1
-        ///     net6.0  => net6.0
-        /// </summary>
-        private string TargetFrameworkToAlias(string target)
```

Comment:
As @NextFreeUsername mentioned, this lacks support for .NET Framework (and others).